trio_run_in_process/run_in_process.py

```diff
@@ -128,67 +94,126 @@ async def _monitor_state(
 @trio_typing.takes_callable_and_args
 async def open_in_process(
     async_fn: Callable[..., TReturn], *args: Any
-) -> AsyncIterator[Process[TReturn]]:
-    proc = Process(async_fn, args)
-
-    parent_r, child_w = os.pipe()
-    child_r, parent_w = os.pipe()
-    parent_pid = os.getpid()
-
-    command = get_subprocess_command(child_r, child_w, parent_pid)
+) -> AsyncIterator[ProcessAPI[TReturn]]:
+    async with open_worker_process() as worker:
```

Comment:
:heart: that `open_in_process` is now just a special case of `open_worker_process`